README.md

```diff
@@ -1,3 +1,148 @@
-# Swish
+# Swish [![Carthage compatible](https://img.shields.io/badge/Carthage-compatible-brightgreen.svg?style=flat-square)](https://github.com/Carthage/Carthage)
 
-Nothing but net(working)
+Nothing but net(working).
+
+Swish is a networking library that is particularly meant for requesting and
+decoding JSON via [Argo](http://github.com/thoughtbot/Argo). It is protocol
+based, and so aims to be easy to test and customize.
+
+## Installation
+
+### [Carthage]
+
+[Carthage]: https://github.com/Carthage/Carthage
+
+Add the following to your Cartfile:
+
+```
+github "thoughtbot/Swish"
+```
+
+Then run `carthage update`.
+
+Follow the current instructions in [Carthage's README][carthage-installation]
+for up to date installation instructions.
+
+[carthage-installation]: https://github.com/Carthage/Carthage#adding-frameworks-to-an-application
+
+## Basic Usage
+```swift
+import Swish
+
+func baseRequest(url url: String, method: RequestMethod) {
+  let url = NSURL(string: url)!
+  let request = NSMutableURLRequest(URL: url)
+
+  request.HTTPMethod = method.rawValue
+}
+
+struct CommentRequest: Request {
+  typealias ResponseType = Comment
+
+  let id: Int
+
+  func build() -> NSURLRequest {
+    let endpoint = "http://example.com/api/comment/\(id)"
+
+    return baseRequest(endpoint: endpoint, method: .GET)
+  }
+}
+
+struct Comment: Decodable {
+  let id: Int
+  let text: String
+  let username: String
+ 
+  static func decode(j: JSON) -> Decoded<Comment> {
+    return curry(Comment.init)
+      <^> j <| "id"
+      <*> j <| "commentText"
+      <*> j <| "username"
+  }
+}
+
+let request = CommentRequest(id: 1)
+APIClient().performRequest(request) { response in
+  // let response: Result<Comment, NSError>
+}
+```
+
+We declare a struct that conforms to the `Request` protocol, which forces us to:
+
+* Declare a `ResponseType` that must be `Decodable`, which allows Swish to use
```

Comment:
Yeah, saw that after. I think the word "must" is probably a bit more prescriptive that intended.